### PR TITLE
ci: run unit tests in chrome and firefox

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2727256](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2727256)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1868/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/1868/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1868/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1868/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1868/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1868/playground/)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -244,8 +244,8 @@ jobs:
           key: ${{ runner.os }}-node-${{ needs.install-deps.outputs.node-version }}-modules-${{ hashFiles('package-lock.json') }}
       - name: Code coverage
         run: |
-          npx nx affected --target=test --browsers=ChromeHeadlessNoSandbox --code-coverage --parallel=3 --progress=false --source-map --watch=false
-          npx nx affected --target=posttest --browsers=ChromeHeadlessNoSandbox --parallel=3 --watch=false
+          npx nx affected --target=test --parallel=3 -c ci
+          npx nx affected --target=posttest --parallel=3 -c ci
```
Comment (johnhwhite) [inline .github/workflows/ci.yml:248]: This approach normalizes the call between Karma and Jest test runners, moving runner-specific parameters to the project configuration.

File: libs/components/core/src/lib/modules/shared/number-format/number-format-utility.spec.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ describe('Number format utility', function () {
     const digits = '0.9-0';
 
     expect(() => formatCurrency(50, digits)).toThrowError(
-      'minimumIntegerDigits value is out of range.'
+      /^(minimumIntegerDigits value is out of range\.|invalid digits value: 0)$/
```
Comment (johnhwhite) [inline libs/components/core/src/lib/modules/shared/number-format/number-format-utility.spec.ts:58]: Firefox has different text in its error message; the error comes from [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat).

File: libs/components/layout/src/lib/modules/text-expand/text-expand.component.spec.ts [hunk 3]
```diff
@@ -490,7 +490,7 @@ describe('Text expand component', () => {
 
       expect(modal).not.toBeNull();
       expect(modalHeader?.innerText.trim()).toBe('Expanded view');
-      expect(modalContent?.innerText.trim()).toBe(expandedText);
+      expect(modalContent?.textContent?.trim()).toBe(expandedText);
```
Comment (johnhwhite) [inline libs/components/layout/src/lib/modules/text-expand/text-expand.component.spec.ts:493]: Firefox collapses space differently.

File: libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.spec.ts [hunk 4]
```diff
@@ -1425,7 +1425,9 @@ describe('Text editor', () => {
         'rgb(51, 51, 51)'
       );
       expect(style.getPropertyValue('color')).toEqual('rgb(238, 238, 238)');
-      expect(style.getPropertyValue('font-family')).toEqual(`"${font}"`);
+      expect(
+        style.getPropertyValue('font-family').replace(/^"|"$/g, '')
+      ).toEqual(font);
```
Comment (johnhwhite) [inline libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.spec.ts:1430]: Firefox does not include the wrapping quotes when retrieving the value.

**Comment by johnhwhite** [comment]
Tests are flakier in Firefox. Closing this PR.

**Pull request closed without merging**